### PR TITLE
feat: allow to define the ports to open when podifying (#1017)

### LINKED ISSUE #1017: When "podifying", I should be able to define the ports I want to open
### Is your enhancement related to a problem? Please describe

As a user, I would like to define which ports from my containers I want to expose.

### Describe the solution you'd like

Adding the ability to select the ports to be opened from the selected containers

Capture d’écran 2022-12-12 à 10 28 08

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

it allow users to define the ports to open when podifying 

### Screenshot/screencast of this PR

expose-port-pod

### What issues does this PR fix or reference?

it fixes #1017 

### How to test this PR?

1. select 1+ containers with ports
2. deselect all ports or some of them
3. create a pod
4. verify that the ports you deselected are not opened

What should it happen when the user selects two containers with the same port? Only one container is displayed in the list and an error message explains what happened?? Only show the warning??

@mairin as i added a new part I also updated the rest with your design https://github.com/containers/podman-desktop/issues/2183 . As the port list is a new stuff, let me know if that's ok or i need to change it

### CODE REVIEW

File: packages/renderer/src/lib/pod/PodCreateFromContainers.svelte [hunk 1]
```diff
@@ -12,7 +12,7 @@ import ContainerIcon from '../images/ContainerIcon.svelte';
 
 let podCreation: PodCreation;
 let createInProgress = false;
-let createError = undefined;
+$: createError = undefined;
```
Comment (benoitf) [inline packages/renderer/src/lib/pod/PodCreateFromContainers.svelte]: It's not clear to me why it's in reactive form ?

Comment (lstocchi) [reply packages/renderer/src/lib/pod/PodCreateFromContainers.svelte]: old stuff, reverted

File: packages/renderer/src/lib/container/ContainerDetailsSummary.svelte [hunk 2]
```diff
@@ -3,6 +3,16 @@ import type { ContainerInfoUI } from './ContainerInfoUI';
 import { getPanelDetailColor } from '../color/color';
 
 export let container: ContainerInfoUI;
+
+function printPorts(ports: number[]): string {
```
Comment (jeffmaury) [inline packages/renderer/src/lib/container/ContainerDetailsSummary.svelte]: This kind of function should be in a utility and not in Svelte/UI

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by slemeur** [comment]
👍

**Comment by mairin** [comment]
@lstocchi The port list looks good! Just a couple of things:

- For the input field labels - can you bold them?

- Can I ask about the name for the pod? I think it always defaults to "my-pod" even if you already have a "my-pod", so on the 2nd time you get an error. Is there a way to add a check and append a -01 / -02 / -03 ... / -$N on the end of the name?

**Comment by lstocchi** [comment]
@mairin 

updated the PR. Now it looks like 
image

is it ok? Let me know, thanks!! :)

Regarding your second point, i'll open a new issue ( https://github.com/containers/podman-desktop/issues/2278 ) so we can work on it in a separate PR. This one is just for defining pod ports. I included a design refactoring bc i added a new element but i don't want to mix too many things.

**Comment by mairin** [comment]
@lstocchi Looks great!!! Thanks for opening the other issue.

**Pull request merged**